docs/fpc-registration-new.puml

```diff
@@ -0,0 +1,116 @@
+@startuml
+
+'- force wrapping of longer text
+'skinparam wrapWidth 400
+
+!pragma teoz true
+
+hide footbox
+title Foot Box removed
+
+title Fabric Private Chaincode - Enclave registry - New approach
+
+actor	"Admin_Org1" as Admin order 10
+participant "ECC_Manager (syscc) Peer1_Org1" as ECC_Manager order 20
+participant "ECC Peer1_Org1" as ECC order 30
+
+note right ECC
+  ECC can be **integrated** in ECC or as **seperate** chaincode
+  Integrated:
+    - ECC and ECC share endorsement policy
+    - attestations stored ECC namespace
+  Separate:
+    - flexible endorsement policies (ECC and ECC can have different)
+    - attestations stored in ECC namespace;
+end note
+
+participant "TLCC_Enclave Peer1_Org1" as TLCC_Enclave order 40 #99FF99 
+
+entity IAS order 50
+
+participant "ECC Peer1_Org2" order 60
+participant "ECC Peer1_Org3" order 61
+
+entity Orderer order 70
+
+group enclave creation and attestation generation
+
+  Admin -> ECC_Manager : create (CC_name, CC_ver, CH_id)
+  activate ECC_Manager
+
+  ECC_Manager -> ECC_Manager : get SPID and IAS credentials from config
+
+  ECC_Manager -> ECC : cc2cc.create(SPID)
+  activate ECC
+
+  create participant ECC_Enclave order 30 #99FF99 
+  ECC -> ECC_Enclave : create (SPID, CC_name, CC_ver, Org_i)
+  activate ECC_Enclave
+
+  ECC_Enclave -> ECC_Enclave : Generate ECC_{S_PK, S_SK, E_PK, E_SK}
+
+  ECC_Enclave -> ECC : Binding(ECC_{S_PK, E_PK})
+  ECC --> ECC_Enclave : Sig_Org_1(ECC_{S_PK, E_PK})
+
+  ECC_Enclave -> TLCC_Enclave : Binding(ECC_{S_PK, E_PK}) using local attestation
+  note over ECC_Enclave, TLCC_Enclave: establish secure channel with TLCC_Enclave
+  TLCC_Enclave --> ECC_Enclave : TLCC_MR, TLCC_{S_PK, E_PK}, CH_id
+  
+  ECC_Enclave -> ECC_Enclave : Generate quote
+  note right
+    Quote includes ... 
+    SPID, CC_name, CC_ver, Org_i,
+    ECC_{S_PK, E_PK}, TLCC_MR,
+    TLCC_{S_PK, E_PK}, CH_id
+  end note
+
+  ECC_Enclave -> ECC_Enclave : Seal state
```

Comment:
we might want to mention explicity what state is?  Presumably (at least), ECC_{S,E}_SK, TLCC_MRENCLAVE, CH_id?